frontend/src/concepts/pipelines/apiHooks/mlmd/__tests__/useGetMlmdContextType.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { testHook, standardUseFetchState } from '~/__tests__/unit/testUtils/hooks';
+import { MetadataStoreServicePromiseClient, ContextType } from '~/third_party/mlmd';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { MlmdContextTypes } from '~/concepts/pipelines/apiHooks/mlmd/types';
+import { useGetMlmdContextType } from '~/concepts/pipelines/apiHooks/mlmd/useGetMlmdContextType';
+import { GetContextTypeResponse } from '~/third_party/mlmd/generated/ml_metadata/proto/metadata_store_service_pb';
+
+// Mock the usePipelinesAPI hook
+jest.mock('~/concepts/pipelines/context', () => ({
+  usePipelinesAPI: jest.fn(),
+}));
+
+// Mock the MetadataStoreServicePromiseClient
+jest.mock('~/third_party/mlmd', () => {
+  const originalModule = jest.requireActual('~/third_party/mlmd');
+  return {
+    ...originalModule,
+    MetadataStoreServicePromiseClient: jest.fn().mockImplementation(() => ({
+      getContextType: jest.fn(),
+    })),
+    GetContextTypeRequest: originalModule.GetContextTypeRequest,
+  };
+});
+
+describe('useGetMlmdContextType', () => {
+  const mockClient = new MetadataStoreServicePromiseClient('');
+  const mockUsePipelinesAPI = jest.mocked(
+    usePipelinesAPI as () => Partial<ReturnType<typeof usePipelinesAPI>>,
+  );
+  const mockGetContextType = jest.mocked(mockClient.getContextType);
+
+  const mockContextType = new ContextType();
+  mockContextType.setId(1);
+  mockContextType.setName('contextType1');
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockUsePipelinesAPI.mockReturnValue({
+      metadataStoreServiceClient: mockClient,
+    });
+  });
+
+  it('should not be loaded when no type is provided', async () => {
+    const renderResult = testHook(useGetMlmdContextType)();
+
+    expect(renderResult.result.current).toStrictEqual(
+      standardUseFetchState(null, false, undefined),
+    );
+  });
```

Comment:
Need to wait for an update then assert you have an error with message `No context type`.

Reply:
its a not ready error so it does not propagate

Reply:
oic now